tensorflow/lite/kernels/activations.cc

```diff
@@ -546,195 +546,78 @@ TfLiteStatus SigmoidEval(TfLiteContext* context, TfLiteNode* node) {
-                             TfLiteSoftmaxParams* params, OpData* data) {
-  const int batch_size = input->dims->data[0];
-  const int intermediate_size = input->dims->data[1];
-  const int input_size = input->dims->data[2];
-  SoftmaxParams op_params;
-  op_params.input_multiplier = data->input_multiplier;
-  op_params.input_left_shift = data->input_left_shift;
-  op_params.diff_min = data->diff_min;
-  optimized_ops::Softmax(
-      op_params, GetTensorShape({batch_size, intermediate_size, 1, input_size}),
-      GetTensorData<uint8_t>(input),
-      GetTensorShape({batch_size, intermediate_size, 1, input_size}),
-      GetTensorData<uint8_t>(output));
-}
-
-// Takes a 4D tensor and perform softmax along the forth dimension.
-void Softmax4DFloat(const TfLiteTensor* input, TfLiteTensor* output,
-                    TfLiteSoftmaxParams* params) {
-  SoftmaxParams op_params;
-  op_params.beta = params->beta;
-  optimized_ops::Softmax(op_params, GetTensorShape(input),
-                         GetTensorData<float>(input), GetTensorShape(output),
-                         GetTensorData<float>(output));
-}
-
-void Softmax4DQuantizedUint8(const TfLiteTensor* input, TfLiteTensor* output,
-                             TfLiteSoftmaxParams* params, OpData* data) {
-  SoftmaxParams op_params;
-  op_params.input_multiplier = data->input_multiplier;
-  op_params.input_left_shift = data->input_left_shift;
-  op_params.diff_min = data->diff_min;
-  optimized_ops::Softmax(op_params, GetTensorShape(input),
-                         GetTensorData<uint8_t>(input), GetTensorShape(output),
-                         GetTensorData<uint8_t>(output));
-}
-
-// TODO(jianlijianli): Try merging Softmax<n>DQuantizedInt8 with
-// Softmax<n>DQuantized, which needs a larger refactor.
-void Softmax1DQuantizedInt8(const TfLiteTensor* input, TfLiteTensor* output,
-                            TfLiteSoftmaxParams* params, OpData* data) {
-  const int input_size = input->dims->data[0];
-  SoftmaxParams op_params;
-  op_params.input_multiplier = data->input_multiplier;
-  op_params.input_left_shift = data->input_left_shift;
-  op_params.diff_min = data->diff_min;
-  optimized_integer_ops::Softmax(
-      op_params, GetTensorShape({1, 1, 1, input_size}),
-      GetTensorData<int8_t>(input), GetTensorShape({1, 1, 1, input_size}),
-      GetTensorData<int8_t>(output));
-}
-
-void Softmax2DQuantizedInt8(const TfLiteTensor* input, TfLiteTensor* output,
-                            TfLiteSoftmaxParams* params, OpData* data) {
-  const int batch_size = input->dims->data[0];
-  const int input_size = input->dims->data[1];
-  SoftmaxParams op_params;
-  op_params.input_multiplier = data->input_multiplier;
-  op_params.input_left_shift = data->input_left_shift;
-  op_params.diff_min = data->diff_min;
-  optimized_integer_ops::Softmax(op_params,
-                                 GetTensorShape({batch_size, 1, 1, input_size}),
```

Comment:
Your code essentially replaces the `GetTensorShape({batch_size, 1, 1, input_size})` call with `GetTensorShape(input)`. However they aren't equal and would result in different shapes being passed to the underlying `optimized_integer_ops::Softmax` function.

Have you verified this using the tests?

Reply:
@nutsiepully : Yes it is verified. All the test cases pass.

Reply:
@ANSHUMAN87 Like I mentioned the code change will lead to different parameters being passed to the underlying function. Could you please address that.

Reply:
@nutsiepully :  That's the idea for the optimization, reuse the existing parameters, and current kernel is capable of handling that. I cant see any issue in that. Would you please describe if you are foreseeing any issue in the change, Thanks!

Reply:
@ANSHUMAN87 Fair point. I didn't look into the implementation of the Softmax.